src/nodes/materials/NodeMaterial.js

```diff
@@ -423,6 +427,38 @@ class NodeMaterial extends Material {
 
 		}
 
+		// ENV MAP
+
+		if ( this.environment === false ) {
```

Comment:
I think we could add this operation in `BasicLighthingModel` removing `this.environment`, and extends `PhongLithingModel` from `BasicLightingModel`

Reply:
Okay, I rewritten the code like that!

One thing that can be optimized though (ideally with a different PR) is that `NodeMaterial.setupLights()` should only add an instance of `EnvironmentNode` for PBR materials. All other materials which do not support PMREM  are incompatible with that type of node. Imo, it seems more clean if this node isn't created for that materials in the first place.

Besides, after looking closer to the implementation of `NodeMaterial`, it would be maybe nicer if the environment related code is not part of `BasicLightingModel`. Right now, it's necessary to access the env node and the combine property in its implementation. Maybe it would be nicer to make a similar type of node like `EnvironmentNode` and provide the result of the environment map sampling as some sort of light input (irradiance) which is then used in `BasicLightingModel`.

Reply:
I made a small revision. The idea of ​​the `LightingModel` parameters would be equivalent to the defines that enable the specific modules of the light model, normally boolean, since we could access the material in the setup process.

Still, I have to do another review, ideally we should have a node like `EnvironmentNode` to handle this.

Reply:
Awesome, that looks much cleaner now!

Reply:
> ideally we should have a node like EnvironmentNode to handle this.

It would be nice if `NodeMaterial` could somehow setup a different type of `EnvironmentNode` depending on the type of material. That would automatically solve https://github.com/mrdoob/three.js/pull/28798#discussion_r1665433550.

Reply:
> Still, I have to do another review, ideally we should have a node like EnvironmentNode to handle this.

I done this part! :)

Reply:
Great! Let's merge the current state. We can add more revisions in other PRs.